### PR TITLE
vendor: updated jwt-go to 3.0.0. Also fixed a few go vet issues.

### PR DESCRIPTION
**Pull request by leetal**
This is in reply to #229.

**WIP:** This uses the current HEAD ref in .glide for Fosite. This MUST be changed to a semantic version of Fosite prior to accepting this PR.

_**NOTE:** Untested_

### CODE REVIEW

File: oauth2/introspector_test.go [hunk 1]
```diff
@@ -62,36 +63,36 @@ var localWarden = &warden.LocalWarden{
...
-	ar2.GrantedScopes = fosite.Arguments{"core"}
+	ar2.GrantedScopes = fosite.Arguments{0: "core"}
 	ar2.RequestedAt = now
 	ar2.Session.(*oauth2.Session).Extra = map[string]interface{}{"foo": "bar"}
 	ar2.Client = &fosite.DefaultClient{ID: "siri"}
 	fositeStore.CreateAccessTokenSession(nil, tokens[1][0], ar2)
 
 	ar3 := fosite.NewAccessRequest(oauth2.NewSession("siri"))
-	ar3.GrantedScopes = fosite.Arguments{"core"}
+	ar3.GrantedScopes = fosite.Arguments{0: "core"}
```
Comment (arekkas) [inline oauth2/introspector_test.go]: this looks weird?

Comment (leetal) [inline oauth2/introspector_test.go]: yeah, that is due to go vet complaining about "composite literal uses unkeyed fields". The only thing that 0 does is silencing that go vet error. Looks really ugly, but does the job.

Comment (arekkas) [inline oauth2/introspector_test.go]: ah right, that's the problem I have with go vet because `gofmt -s` actually removes those

Comment (leetal) [inline oauth2/introspector_test.go]: Yeah ;) I think i'll remove the 0's. Will just confuse lesser bewildered gophers.

**Comment by arekkas** [comment]
ah right, the error is because the client does not have the privilege to access the warden api which was previously possible due to my mistake which @faxal solved here: https://github.com/ory-am/fosite/commit/7faee6bbd53ee762ddfe194fb2ea5e7d0205e46d

this is now #232

**Pull request closed without merging**